src/main.go

```diff
@@ -30,6 +30,28 @@ func main() {
 		c.JSON(http.StatusOK, gin.H{"id": intID})
 	})
 
+	//　練習課題2.1
+	router.POST("/", func(c *gin.Context) {
+		var json struct {
+			Message string "json:\"message\""
+		}
+		c.JSON(http.StatusCreated, gin.H{"message": json.Message})
```

Comment:
POSTされたjsonデータを変数`json`に展開するコードが含まれていないようです
以下のように実装するのはどうでしょうか
```suggestion
		err := c.ShouldBindJSON(&json)
		if err != nil {
			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
			return
		}
		c.JSON(http.StatusCreated, gin.H{"message": json.Message})
```
`c.ShouldBindJSON(&json)`はGoの構造体のポインタを受け取り、その構造体の中身にPOSTされてきた内容（実はここの変数`c`にすべて含まれています）を適切に展開するメソッドです。
ポインタとかよくわからないと思うのですが、まあこう書くと上手くいくんだな～ぐらいに思ってもらえばよいです。